core/internal/filestream/retry_policy.go

```diff
@@ -0,0 +1,43 @@
+package filestream
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/hashicorp/go-retryablehttp"
+)
+
+// RetryPolicy is the retry policy to be used for file stream operations.
+func RetryPolicy(
+	ctx context.Context,
+	resp *http.Response,
+	err error,
+) (bool, error) {
+	// non-retryable status codes
+	if resp != nil {
+		switch resp.StatusCode {
+		case http.StatusBadRequest: // don't retry on 400 bad request or 409 conflict
+			return false, err
+		case http.StatusUnauthorized: // don't retry on 401 unauthorized
+			return false, err
+		case http.StatusForbidden: // don't retry on 403 forbidden
+			return false, err
+		case http.StatusNotFound: // don't retry on 404 not found
+			return false, err
+		case http.StatusConflict: // don't retry on 409 conflict
+			return false, err
+		case http.StatusGone: // don't retry on 410 Gone
+			return false, err
+		}
+	}
+	// if err != nil, retryablehttp's base policy is to retry
+	// if the error is recoverable, meaning it's not:
+	//   - due to too many redirects.
+	//   - due to an invalid protocol scheme.
+	//   - due to an invalid header.
+	//   - due to TLS cert verification failure.
+	// if resp != nil and err == nil, retryablehttp's base policy is to retry
+	// if the status code is 429, 5xx, or invalid. We want to be extra cautious
+	// and retry in other remaining cases as well.
+	return retryablehttp.ErrorPropagatedRetryPolicy(ctx, resp, err)
```

Comment:
idk tbh. they haven't changed that code for 4 years and if they do, we'll see that in the dependabot's vendor update. the alternative is to copy a lot of code from them, which is weird. i left out such a long comment so that the developer doesn't need to click into their code and read it there.

Reply:
My point is that this is an accidental similarity---nothing in the name or the documentation of the function implies the behavior we want. This is generally considered a code smell.

The function doesn't do anything complicated and its source isn't very long. It's better to be explicit about our retry conditions and write out the code here. The comment doesn't prevent a reader from having to click through to their implementation---if I'm debugging an issue, I won't trust a comment!